src/errorhandler.h

```diff
@@ -48,6 +48,14 @@ class ErrorHandler final : public QObject {
   };
   Q_ENUM(AlertType)
 
+  // This can be an boolean but it's better to have an enum to improve the
+  // readability of the code.
+  enum ErrorPropagation {
+    NoErrorPropagation,
+    PropagateError,
+  };
+  Q_ENUM(ErrorPropagation);
```

Comment:
It's weird to me that this enum is defined on the error handler when in fact its being used elsewhere in the code. AFAICT it's only used in the Tasks, what about we move this declaration there?

Reply:
I thought about that, but why tasks? we can have the same enum in every single class or implement it centralized. ErrorHandler seems to be the right choice. Balrog is not a task and still need the ErrorHandler enum

Reply:
Ah! I missed Balrog. Ok yeah, that makes it tricker.

Reply:
Ah, the new commit changes thing in a way that it makes complete sense for the enum to be on the ErrorHandler class.